contracts/RootChain/RootChain.sol

```diff
@@ -109,22 +104,30 @@ contract RootChain {
         payable
     {
         var txList = txBytes.toRLPItem().toList();
-        require(txList.length == 11);
-        for (uint256 i; i < 6; i++) {
+        for(uint256 i = 0; i < 6; i++) {
             require(txList[i].toUint() == 0);
         }
+        require(txList[6].toAddress() == msg.sender);
         require(txList[7].toUint() == msg.value);
-        require(txList[9].toUint() == 0);
```

Comment:
We need to check txList[9] otherwise someone could include more money in the second output than they deposited

Reply:
yep, fixing that